### PR TITLE
BOAC-4112: Alerts user to students already assigned degree check

### PR DESCRIPTION
**Pull request by lyttam**
https://jira.ets.berkeley.edu/jira/browse/BOAC-4112

### CODE REVIEW

File: boac/api/degree_progress_student_controller.py [hunk 1]
```diff
@@ -166,6 +166,30 @@ def get_degree_checks(uid):
...
+@can_edit_degree_progress
+def get_students(template_id):
+    params = request.get_json()
+    sids = get_param(params, 'sids')
+    if not sids:
+        raise BadRequestError('sids is required.')
+    template = DegreeProgressTemplate.find_by_id(template_id)
+    if not template:
+        raise ResourceNotFoundError('Degree template not found')
+    degree_checks = DegreeProgressTemplate.find_by_name(template.degree_name, student_sids=sids)
```
Comment (johncrossman) [inline boac/api/degree_progress_student_controller.py:179]: @lyttam - This won't work if the template is renamed, right? Will we change the logic when student degree-checks have foreign-key ref back to template?  If yes, maybe a `TODO:` line above would remind us later.

Comment (lyttam) [reply boac/api/degree_progress_student_controller.py:179]: You're right @johncrossman - but maybe that's ok? What if the template gets changed (including the name) and the advisor wants all their students to get the new version?

Comment (lyttam) [reply boac/api/degree_progress_student_controller.py:179]: Either way, matching by the degree name doesn't seem ideal. I will add a TODO regardless.

Comment (johncrossman) [reply boac/api/degree_progress_student_controller.py:179]: @lyttam - I think "advisor wants all their students to get the new version" was ruled out as a requirement. We can circle back to this TODO soon.

Comment (lyttam) [reply boac/api/degree_progress_student_controller.py:181]: @johncrossman Thanks, I amended with a TODO comment.

**Comment by johncrossman** [comment]
Thanks @lyttam  Waiting on Travis.

**Comment by johncrossman** [comment]
`lint-py` complains.

**Comment by lyttam** [comment]
Fixed the lint-py issue, @johncrossman

**Pull request merged**